opentaskspal/src/test/java/org/dmfs/opentaskspal/tasks/InstanceDataTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import org.dmfs.optional.Present;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.rfc5545.Duration;
+import org.dmfs.tasks.contract.TaskContract;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.TimeZone;
+
+import static org.dmfs.android.contentpal.testing.contentoperationbuilder.WithValues.withValuesOnly;
+import static org.dmfs.android.contentpal.testing.contentvalues.Containing.containing;
+import static org.dmfs.android.contentpal.testing.contentvalues.NullValue.withNullValue;
+import static org.dmfs.android.contentpal.testing.rowdata.RowDataMatcher.builds;
+import static org.dmfs.optional.Absent.absent;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * @author Marten Gajda
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class InstanceDataTest
+{
+    @Test
+    public void testNoDate() throws Exception
+    {
+        assertThat(new InstanceData(5),
+                builds(
+                        withValuesOnly(
+                                withNullValue(TaskContract.Instances.INSTANCE_START),
+                                withNullValue(TaskContract.Instances.INSTANCE_START_SORTING),
+                                withNullValue(TaskContract.Instances.INSTANCE_DUE),
+                                withNullValue(TaskContract.Instances.INSTANCE_DUE_SORTING),
+                                withNullValue(TaskContract.Instances.INSTANCE_DURATION),
+                                containing(TaskContract.Instances.INSTANCE_ORIGINAL_TIME, 0L),
+                                containing(TaskContract.Instances.INSTANCE_STATUS, 5)
```

Comment:
`INSTANCE_STATUS` is not replaced yet here with `DISTANCE_FROM_CURRENT`.

Reply:
Strange that Travis didn't catch it..

Reply:
apparently it doesn't run these tests.

Reply:
maybe it's because `opentaskspal` doesn't have any Android tests. I've added the `check` task, let's see if that catches the issue.

Reply:
yep, now it's catched. We should check all our Android projects to make sure the `check` task is run.

Reply:
I don't understand this, those unit tests should be part of the build.
Docs even say here that 'check' is the default gradle task:
https://docs.travis-ci.com/user/languages/java/
I tried to google it, haven't found any issue mentioned.
Anyway, it's good that it works now.

Reply:
right, `check` is the default, but we have overridden that in the config to run the connected test

Reply:
oh, I see